ios/KMMIC/Sources/Presentation/Modules/LoginScreen/LoginViewModel.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  LoginViewModel.swift
+//  KMMIC
+//
+//  Created by MarkG on 20/04/2023.
+//  Copyright © 2023 Nimble. All rights reserved.
+//
+
+import Combine
+import Factory
+
+final class LoginViewModel: ObservableObject {
+
+    @Injected(\.loginUseCase) private var loginUseCase
+
+    private var cancellableBag = Set<AnyCancellable>()
+
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var isAllValidated: Bool = false
+    @Published var isLoading: Bool = false
+    @Published var didLogin: Bool = false
+    @Published var alertDescription: AlertDescription?
+
+    init() {
+        Publishers.CombineLatest($email, $password)
+            .map { !$0.0.isEmpty && !$0.1.isEmpty }
+            .assign(to: &$isAllValidated)
+    }
+
+    func login() {
+        isLoading = true
+
+        loginUseCase.callAsPublisher(email: email, password: password)
```

Comment:
Why don't use `callAsFunction`

Reply:
@suho `callAsFunction` is quite confusing for me, what is the call as function mean? 🤔 
`callAsNative` is also call it as a function

Reply:
[callAsFunction](https://github.com/apple/swift-evolution/blob/main/proposals/0253-callable.md) is a feature of Swift. With that, you can use use case like this:
```swift
loginUseCase(email: email, password: password)
```